cmd/client/search.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+
+	"github.com/monstercat/gologx/logxhost"
+	"github.com/monstercat/gologx/utils"
+	"github.com/olekukonko/tablewriter"
+)
+
+func showSearch(name string, args []string) error {
+	var postgres string
+	var service string
+	var matcher string
+	var limit int
+	var dateBefore string
+	var dateAfter string
+	var orderby string
+
+	limitDefault := 50
+
+	set := flag.NewFlagSet(name, flag.ExitOnError)
+	set.StringVar(&postgres, "postgres", "", "Postgres database")
+	set.StringVar(&service, "service", "", "Defines a service to be filtered")
+	set.StringVar(&matcher, "matcher", "", "Defines a matcher to be filtered. Searches machine and name")
+	set.IntVar(&limit, "limit", limitDefault, "Defines the limit of results in the list")
+	set.StringVar(&dateBefore, "datebefore", "", "Filters before a specific date time")
+	set.StringVar(&dateAfter, "dateafter", "", "Filters after a specific date time")
+	set.StringVar(&orderby, "orderby", "", "Orders list by a defined column")
+
+	if err := set.Parse(args); err != nil {
+		return err
+	}
+
+	if service == "" {
+		fmt.Println(utils.StyleDanger("service flag is required"))
+		os.Exit(1)
+	}
+
+	dateRegex := regexp.MustCompile("((19|20)\\d\\d)-(0?[1-9]|1[012])-(0?[1-9]|[12][0-9]|3[01])")
```

Comment:
In the future we should just do time.Parse, and pass in time.Time to the query generator. The sql arguemnts don't need to be strings.